src/reader/parser.rs

```diff
@@ -349,15 +349,15 @@ impl PullParser {
     /// # Parameters
     /// * `t`       --- next token;
     /// * `on_name` --- a callback which is executed when whitespace is encountered.
-    fn read_qualified_name(&mut self, t: Token, target: QualifiedNameTarget,
-                           on_name: |&mut PullParser, Token, OwnedName| -> Option<XmlEvent>) -> Option<XmlEvent> {
+    fn read_qualified_name<'a, F>(&'a mut self, t: Token, target: QualifiedNameTarget, on_name: F) -> Option<XmlEvent> 
+      where F: Fn<(&'a mut PullParser, Token, OwnedName), Option<XmlEvent>> {
```

Comment:
Doesn't seem to work for me :(

    src/reader/parser.rs:353:20: 353:35 error: missing lifetime specifier [E0106]
    src/reader/parser.rs:353       where F: Fn<(&mut PullParser, Token, OwnedName), Option<XmlEvent>> {

So that isn't what the inferred lifetimes would be anyway?